File: TPlaba2_Java/src/Jewelry.java
import java.awt.Color;
import java.awt.Graphics;

public abstract class Jewelry implements Stone {
	protected int srartRosX;
	protected int srartRosY;

	public int Weight;
	public int Price;
	public int Hardness;
	public Color ColorStone;
	
	protected abstract void setWeight(int w);
	
	public abstract int getWeight();
	
	protected abstract void setPrice(int p);
	
	public abstract int getPrice();
	
	protected abstract void setHardness(int h);
	
	public abstract int getHardness();

	public abstract void drawStone(Graphics g);

	public void setPosition(int x, int y) {
		srartRosX = x;
		srartRosY = y;
	}
}

File: WindowsFormsApplicationLab2/WindowsFormsApplicationLaba2/Jewelry.cs
﻿using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WindowsFormsApplicationLaba2
{
    public abstract class Jewelry : Stone
    {
        protected float srartRosX;
        protected float srartRosY;

        public void setPosition(int x, int y)
        {
            srartRosX = x;
            srartRosY = y;
        }

        public abstract void drawStone(Graphics g);

        //вес
        public virtual double Weight { protected set; get; }

        //цена
        public virtual double Price { protected set; get; }

        //твердость
        public virtual int Hardness { protected set; get; }

        //цвет 
        public Color ColorStone { protected set; get; }

        public virtual void setMainColor(Color color)
        {
            ColorStone = color;
        }

    }
}
